libafl/src/generators/mod.rs

```diff
@@ -71,54 +71,47 @@ where
 
 #[derive(Clone, Debug)]
 /// Generates random bytes
-pub struct RandBytesGenerator<S> {
-    max_size: usize,
-    phantom: PhantomData<S>,
+pub struct RandBytesGenerator {
+    max_size: NonZeroUsize,
 }
 
-impl<S> Generator<BytesInput, S> for RandBytesGenerator<S>
+impl<S> Generator<BytesInput, S> for RandBytesGenerator
 where
     S: HasRand,
 {
     fn generate(&mut self, state: &mut S) -> Result<BytesInput, Error> {
-        let mut size = state.rand_mut().zero_upto(self.max_size);
-        if size == 0 {
-            size = 1;
-        }
+        let size = 1 + state.rand_mut().below(self.max_size);
```

Comment:
can wrap around for MAX_USIZE again :P

Reply:
No. self.max_size returns max_usize - 1 at max